types/types.go

```diff
@@ -331,6 +331,14 @@ type ImageDestination interface {
 	Commit(ctx context.Context, unparsedToplevel UnparsedImage) error
 }
 
+type ImageDestinationReusableWithRef interface {
```

Comment:
Let’s at least try to stop making the write side public; define this in `internal/types` or something like that.

RFC: @nalind — is this practical to do with the blob cache layer?

Reply:
Moved this to `internal/types`.